java/main/java/com/couchbase/lite/internal/NativeLibrary.java

```diff
@@ -43,132 +42,162 @@ private NativeLibrary() { }
-            }
+        try { System.load(libPath); }
+        catch (Throwable e) {
+            throw new IllegalStateException(
+                "Failed loading native library " + lib + " @" + libPath + " (" + os + "/" + arch + ")",
+                e);
         }
     }
 
     /**
-     * Calculate the MD5 digest for all of the libraries.
+     * Read the MD5 digest for a library and construct the target directory file from it.
      */
     @NonNull
-    private static BigInteger computeTargetDirectory(
+    private static File computeTargetDirectory(
+        @NonNull File scratchDir,
         @NonNull String resDirPath,
-        @NonNull List<String> libs,
-        @NonNull byte[] buf)
-        throws IOException {
+        @NonNull String lib) {
+        final String path = resDirPath + JAVA_PATH_SEPARATOR + lib + "." + DIGEST_MD5;
+        try (InputStream rezStream = NativeLibrary.class.getResourceAsStream(path)) {
+            if (rezStream == null) { throw new IOException("Cannot find digest resource"); }
+
+            final String hash;
+            try (BufferedReader in = new BufferedReader(new InputStreamReader(rezStream, StandardCharsets.UTF_8))) {
+                hash = in.readLine();
+            }
+            if (hash == null) { throw new IOException("Digest file is empty"); }
 
-        // get a message digest for
-        final MessageDigest md;
-        try { md = MessageDigest.getInstance(DIGEST_MD5); }
-        catch (NoSuchAlgorithmException e) {
-            throw new IllegalStateException("No such digest algorithm: " + DIGEST_MD5, e);
+            return new File(scratchDir, hash.trim()).getCanonicalFile();
         }
-
-        for (String lib: libs) {
-            final String path = resDirPath + JAVA_PATH_SEPARATOR + lib + "." + DIGEST_MD5;
-            try (InputStream in = NativeLibrary.class.getResourceAsStream(path)) {
-                if (in == null) { throw new IOException("Cannot find MD5 for library at " + path); }
-                int bytesRead;
-                while ((bytesRead = in.read(buf)) != -1) { md.update(buf, 0, bytesRead); }
-            }
+        catch (IOException e) {
+            throw new IllegalStateException("Cannot read digest file: " + path, e);
         }
-
-        return new BigInteger(1, md.digest());
     }
 
+    // Per the support matrix: Linux, Windows and Mac (loosy goosy checking)
     @NonNull
     private static String getOsDir(@NonNull String osName) {
         final String os = osName.toLowerCase(Locale.getDefault());
-        if (os.contains("mac")) { return LIB_DIR_MAC; }
-        if (os.contains(LIB_DIR_LINUX)) { return LIB_DIR_LINUX; }
-        if (os.contains(LIB_DIR_WINDOWS)) { return LIB_DIR_WINDOWS; }
-        throw new IllegalStateException("Unrecongnized OS: " + osName);
+        if (os.contains(OS_WINDOWS)) { return OS_WINDOWS; }
+        if (os.contains(OS_LINUX)) { return OS_LINUX; }
+        if (os.contains("mac")) { return OS_MAC; }
```

Comment:
Curious why not contains(OS_MAC) ?

Reply:
Maybe that's better.  Lemme confirm.

Reply:
That won't work.  The Java recommended way of identifying the MacOS is `System.getProperty("os.name").toLowerCase().contains("mac")`.  The directory containing the MacOS libraries is named "macos" (which is the value of OS_MAC).  I'm going to take the sense of your point and introduce a new constant, MAC_OS_DIR, which will contains the name of the directory.  OS_MAC with be "mac".